README.rst

```diff
@@ -454,16 +454,184 @@ template ``context_processors`` in ``settings.py`` as shown below.
+    )
+
+It is recommended to use ``register_menu_group`` in the ``ready`` method of the AppConfig.
+
+Note: ``register_menu_items`` is deprecated by ``register_menu_group`` and will be removed in the
+future versions. ``register_menu_items`` will be shown at the top of navigation menu and above 
+any register_menu_group items.
+
+``Create a general link``
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you want to create a link which contains a custom url then you can use following syntax for config.
+
+**Syntax:**
+
+.. code-block:: python
+
+    my_link_config = {'label': 'Link Label', 'url': 'link_url', 'icon':'my-icon'}
+    register_menu_group(position=1, config=my_link_config)
+
+Following is the description of the configuration:
+
++------------------+--------------------------------------------------------------+
+| **Parameter**    | **Description**                                              |
++------------------+--------------------------------------------------------------+
+| ``label``        | (``str``) Display name for the link.  (``required``)         |
++------------------+--------------------------------------------------------------+
+| ``url``          | (``str``) url of the link.  (``required``)                   |
++------------------+--------------------------------------------------------------+
+| ``icon``         | (``str``) CSS class name of the link icon.  (``optional``)   |
++------------------+--------------------------------------------------------------+
+
+``Create a model link``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If you want to create a link that contains a url of ``add`` or ``list`` page of a model
+then you can use following syntax. Users will only be able to see links for 
+models they have permission to either view or edit.
+
+**Syntax:**
+
+.. code-block:: python
+
+    # use to create a link of list page
+    my_model_list_config = {
+        'model': 'my_project.MyModel',
+        'name': 'changelist',
+        'label': 'MyModel Change List',
+        'icon': 'my-model-list-class',
+    }
+
+    #use to create a link of add page
+    my_model_add_config = {
+        'model': 'my_project.MyModel',
+        'name': 'add',
+        'label': 'MyModel Add Item',
+        'icon': 'my-model-add-class',
+    }
+    register_menu_group(position=1, config=my_model_list_config)
+    register_menu_group(position=2, config=my_model_add_config)
+
+Following is the description of the configuration:
+
++------------------+--------------------------------------------------------------+
+| **Parameter**    | **Description**                                              |
++------------------+--------------------------------------------------------------+
+| ``model``        | (``str``) Model of a app.  (``required``)                    |
++------------------+--------------------------------------------------------------+
+| ``name``         | (``str``) url name. eg. changelist or add.  (``required``)   +
```

Comment:
are you sure that plus sign at the end is right?

Reply:
> Hi,
> 
> I pulled this again and tried, but this is not fixed: [#175 (review)](https://github.com/openwisp/openwisp-utils/pull/175#pullrequestreview-670415003)
> 
> This PR was in ready for review tab,

It's working correctly. I fixed it last time. for icons, let's do it later in https://github.com/openwisp/openwisp-utils/issues/206 or https://github.com/openwisp/openwisp-utils/pull/185.